pkg/rook/purge_osd.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rook
+
+import (
+	ctx "context"
+	"fmt"
+	"log"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/bash"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/mons"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func PurgeOsd(context *k8sutil.Context, operatorNamespace, clusterNamespace, osdId, flag string) string {
+
+	monCm, err := context.Clientset.CoreV1().ConfigMaps(clusterNamespace).Get(ctx.TODO(), mons.MonConfigMap, v1.GetOptions{})
+	if err != nil {
+		log.Fatalf("failed to get mon configmap %s %v", mons.MonConfigMap, err)
+	}
+
+	monEndPoint := monCm.Data["data"]
+	cephArgs := []string{
+		"auth", "print-key", "client.admin",
+	}
+	adminKey := bash.RunCommandInOperatorPod(context, "ceph", cephArgs, operatorNamespace, clusterNamespace)
```

Comment:
The client.admin key should already be found in a file in the operator pod. Can we just reference that file instead of querying ceph for it?

Reply:
it fails with ```rookcmd: failed to read ceph secret file from "/var/lib/rook-ceph-mon": open /var/lib/rook-ceph-mon/secret.keyring: no such file or directory``` and it come from https://github.com/rook/rook/blob/11e80b7047553c04eeeb5c11f1c6c76fda381d5e/cmd/rook/ceph/osd.go#L369-L386

it looks for file `"/var/lib/rook-ceph-mon/secret.keyring"` but I checked it is not present inside rook op pod